File: typescript/src/resources/send.ts
import type { HttpClient } from "../http"
import type { SendParams, SendResult } from "../types"

export class Send {
  constructor(private http: HttpClient) {}

  /** Send an email */
  async email(params: SendParams): Promise<SendResult> {
    return this.http.post<SendResult>("/send", params)
  }
}

File: python/sentroy/send.py
from __future__ import annotations

from sentroy._http import _HttpClient
from sentroy.types import SendParams, SendResult


class SendResource:
    """Interact with the Send API."""

    def __init__(self, http: _HttpClient) -> None:
        self._http = http

    def email(self, params: SendParams) -> SendResult:
        """Send an email."""
        data = self._http.post("/send", params.to_dict())
        return SendResult.from_dict(data)
